src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs

```diff
@@ -2106,28 +2107,28 @@ private void AddCustomAttributesToTable<T>(IEnumerable<T> parentList, Func<T, En
             }
         }
 
-        protected virtual int AddCustomAttributesToTable(EntityHandle parentHandle, IEnumerable<ICustomAttribute> attributes)
+        protected virtual void AddCustomAttributesToTable(EntityHandle parentHandle, IEnumerable<ICustomAttribute> attributes)
         {
-            int count = 0;
             foreach (var attr in attributes)
             {
-                count++;
                 AddCustomAttributeToTable(parentHandle, attr);
             }
-            return count;
         }
 
-        private void AddCustomAttributeToTable(EntityHandle parentHandle, ICustomAttribute customAttribute)
+        protected bool AddCustomAttributeToTable(EntityHandle parentHandle, ICustomAttribute customAttribute)
         {
             IMethodReference constructor = customAttribute.Constructor(Context, reportDiagnostics: true);
-
             if (constructor != null)
             {
                 metadata.AddCustomAttribute(
                     parent: parentHandle,
                     constructor: GetCustomAttributeTypeCodedIndex(constructor),
                     value: GetCustomAttributeSignatureIndex(customAttribute));
+
+                return true;
             }
+
+            return false;
```

Comment:
>return false [](http://example.com/codeflow?start=12&length=12)

Are we testing the EnC case where the attribute constructor is missing?

Reply:
I don't think we are. I can add it.

Reply:
Actually, I don't understand how we can get into situation where we do not have the constructor (not just for EnC, but in general). The method only returns null if the attribute type is a struct with default ctor:

https://github.com/dotnet/roslyn/blob/main/src/Compilers/CSharp/Portable/Emitter/Model/AttributeDataAdapter.cs#L49

But then that's an error, so we won't make it to emit phase.

Seems like we should assert here that the constructor is never null.

Reply:
In any case, I'd leave that for another PR.